app/main.py

```diff
@@ -1,3 +1,69 @@
-def shop_trip():
-    # write your code here
-    pass
+from decimal import Decimal, ROUND_HALF_UP
+import json
+
+from app.customer import Customer
+from app.shop import Shop
+
+
+with open("app/config.json", "r") as json_file:
+    data_dict = json.load(json_file)
+
+FUEL_PRICE = Decimal(data_dict["FUEL_PRICE"])
+
+customers = [Customer(*data_dict["customers"][i].values())
+             for i in range(len(data_dict["customers"]))]
+
+shops = [Shop(*data_dict["shops"][i].values())
+         for i in range(len(data_dict["shops"]))]
+
+
+def shop_trip() -> None:
+    for customer in customers:
+        print(f"{customer.name} has {customer.money} dollars")
+
+        min_milk_cost = Decimal("0")
+        min_bread_cost = Decimal("0")
+        min_butter_cost = Decimal("0")
+        min_total_cost = Decimal("Infinity")
+        min_shop = None
+
+        for shop in shops:
+            distance = Decimal((((shop.location[0] - customer.location[0]) ** 2
+                                 + (shop.location[1]
+                                    - customer.location[1]) ** 2) ** 0.5))
+            fuel_cost = (distance * Decimal(customer.car["fuel_consumption"])
+                         / Decimal("100") * FUEL_PRICE) * 2
+            milk_cost = (Decimal(str(customer.product_cart["milk"]))
+                         * Decimal(str(shop.products["milk"])))
+            bread_cost = (Decimal(str(customer.product_cart["bread"]))
+                          * Decimal(str(shop.products["bread"])))
+            butter_cost = (Decimal(str(customer.product_cart["butter"]))
+                           * Decimal(str(shop.products["butter"])))
```

Comment:
Move serving customer logic to a shop class. You have a lot going on in this function, make sure to follow SRP